pkg/gather/ssh/ssh.go

```diff
@@ -141,3 +191,84 @@ func LoadPrivateSSHKeys(paths []string) (map[string]interface{}, error) {
 	}
 	return keys, nil
 }
+
+// newHTTPProxy creates a new httpProxy populated with credentials
+// if any is available
+func newHTTPProxy(uri *url.URL, forward proxy.Dialer) (proxy.Dialer, error) {
+	proxyDialer := new(httpProxy)
+	proxyDialer.host = uri.Host
+	proxyDialer.forward = forward
+	if uri.User != nil {
+		proxyDialer.haveAuth = true
+		proxyDialer.username = uri.User.Username()
+		proxyDialer.password, _ = uri.User.Password()
+	}
+
+	return proxyDialer, nil
+}
+
+// newHTTPSProxy creates a new httpProxy populated with credentials using HTTPS dialer
+func newHTTPSProxy(uri *url.URL, forward proxy.Dialer) (proxy.Dialer, error) {
+	return newHTTPProxy(uri, HttpsDialer)
+}
+
+// Dial is a custom dialer for the ssh client that will use HTTP CONNECT method via TLS handshake
+func (d httpsDialer) Dial(network, addr string) (c net.Conn, err error) {
+	c, err = tls.Dial("tcp", addr, TlsConfig)
+	if err != nil {
+		fmt.Println(err)
+	}
+	return
```

Comment:
the default return should probably be:

`return c, nil`

Reply:
done